File: frontend/src/types/theme.ts
// Theme types mirrored from backend src/schemas/theme.py

export interface ThemeOut {
  id: number;
  slug: string;
  name_en: string;
  name_pt: string;
  parent_id: number | null;
  sort_order: number;
}

export interface ThemeTreeOut extends ThemeOut {
  children: ThemeTreeOut[];
}

File: backend/src/schemas/theme.py
"""Theme schemas."""

from typing import Optional

from pydantic import BaseModel


class ThemeOut(BaseModel):
    """Flat theme output."""

    model_config = {"from_attributes": True}

    id: int
    slug: str
    name_en: str
    name_pt: str
    parent_id: int | None = None
    sort_order: int


class ThemeTreeOut(ThemeOut):
    """Hierarchical theme with children."""

    children: list["ThemeTreeOut"] = []


# Forward ref resolved
ThemeTreeOut.model_rebuild()
